deploy/buildtools/buildfarmhostdispatcher.py

```diff
@@ -0,0 +1,302 @@
+        BuildFarmHostDispatcher.__init__(self, build_config, args)
+
+        self.launched_instance_object = None # type: ignore
+        self.instance_type = ""
+        self.build_instance_market = ""
+        self.spot_interruption_behavior = ""
+        self.spot_max_price = ""
+
+        # ec2 hosts will never be local
+        self.is_local = False
+
+    def parse_args(self) -> None:
+        """Parse build farm host arguments."""
+        # get aws specific args
+        self.instance_type = self.args['instance-type']
+        self.build_instance_market = self.args['build-instance-market']
+        self.spot_interruption_behavior = self.args['spot-interruption-behavior']
+        self.spot_max_price = self.args['spot-max-price']
+
+        self.override_remote_build_dir = self.args.get("build-dir")
+
+    def request_build_farm_host(self) -> None:
+        """Launch an AWS EC2 instance for the build config."""
+        build_instance_market = self.build_instance_market
+        spot_interruption_behavior = self.spot_interruption_behavior
+        spot_max_price = self.spot_max_price
+
+        num_instances = 1
+        self.launched_instance_object = EC2BuildFarmHostDispatcher.ec2_launch_instances(
+            self.instance_type,
+            num_instances,
+            build_instance_market,
+            spot_interruption_behavior,
+            spot_max_price)[0]
+
+    @staticmethod
+    def request_build_farm_hosts(build_farm_host_dispatchers: List['EC2BuildFarmHostDispatcher']) -> None: # type: ignore[override]
+        """Launch multiple AWS EC2 instances for the build configs."""
+        # TODO: this can be further optimized (spawn-like build farm hosts)
+        # double-check that build farms are the same
+        it_len = len(set([x.instance_type for x in build_farm_host_dispatchers]))
+        bim_len = len(set([x.build_instance_market for x in build_farm_host_dispatchers]))
+        sib_len = len(set([x.spot_interruption_behavior for x in build_farm_host_dispatchers]))
+        smp_len = len(set([x.spot_max_price for x in build_farm_host_dispatchers]))
+        if it_len == 1 and bim_len == 1 and sib_len == 1 and smp_len == 1:
+            amt_requested = len(build_farm_host_dispatchers)
+            instance_type = build_farm_host_dispatchers[0].instance_type
+            build_instance_market = build_farm_host_dispatchers[0].build_instance_market
+            spot_interruption_behavior = build_farm_host_dispatchers[0].spot_interruption_behavior
+            spot_max_price = build_farm_host_dispatchers[0].spot_max_price
+
+            launched_objs = EC2BuildFarmHostDispatcher.ec2_launch_instances(
+                instance_type,
+                amt_requested,
+                build_instance_market,
+                spot_interruption_behavior,
+                spot_max_price)
+
+            for bh, lo in zip(build_farm_host_dispatchers, launched_objs):
+                bh.launched_instance_object = lo
+        else:
```

Comment:
Given that `awstools.launch_instances` breaks this down into individual AWS instance launches, all of this extra optimization seems like it might be extra complexity that doesn't buy us anything but more code to maintain.  Am I missing something?

Reply:
Good catch. I'll go ahead and remove the optimization.